File: questions/1657. Determine if Two Strings Are Close/determine.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        vector<int> a1(26,0), a2(26,0);
        set<char> s1, s2;
        for(int i = 0; i < word1.size(); i++){
            a1[(int)(word1[i] - 'a')]++;
            s1.insert(word1[i]);
        }
        for(int i = 0; i < word2.size(); i++){
            a2[(int)(word2[i] - 'a')]++;
            s2.insert(word2[i]);
        }
        sort(a1.begin(), a1.end());
        sort(a2.begin(), a2.end());
        return s1==s2 && a1==a2;
    }
};

File: questions/1657. Determine if Two Strings Are Close/determine.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        int[] count1 = new int[26];
        int[] count2 = new int[26];
        HashSet<Character> set1 = new HashSet();
        HashSet<Character> set2 = new HashSet();
        for(char a: word1.toCharArray()) {count1[(int)(a - 'a')]++; set1.add(a);}
        for(char a: word2.toCharArray()) {count2[(int)(a - 'a')]++; set2.add(a);}

        Arrays.sort(count1);
        Arrays.sort(count2);
        
        return Arrays.equals(count1, count2) && set1.equals(set2);
    }
}
